sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -299,6 +304,202 @@ object OpenNumCountedTestInputStream extends Logging {
     }
   }
 }
+class RocksDBStateEncoderSuite extends SparkFunSuite {
+
+  // Helper method to create test schemas
+  private def createTestSchemas() = {
+    val keySchema = StructType(Seq(
+      StructField("k1", IntegerType),
+      StructField("k2", LongType),
+      StructField("k3", DoubleType)
+    ))
+    val valueSchema = StructType(Seq(
+      StructField("v1", StringType),
+      StructField("v2", BooleanType)
+    ))
+    (keySchema, valueSchema)
+  }
+
+  // Create encoders for different key encoding strategies
+  private def createTestEncoder(keyStateEncoderSpec: KeyStateEncoderSpec): RocksDBDataEncoder = {
+    val (keySchema, valueSchema) = createTestSchemas()
+    val stateSchemaInfo = Some(StateSchemaInfo(
+      keySchemaId = 0,
+      valueSchemaId = 0
+    ))
+    new AvroStateEncoder(keyStateEncoderSpec, valueSchema, stateSchemaInfo)
+  }
+
+  private def createNoPrefixKeyEncoder(): RocksDBDataEncoder = {
+    val (keySchema, _) = createTestSchemas()
+    createTestEncoder(NoPrefixKeyStateEncoderSpec(keySchema))
+  }
+
+  private def createPrefixKeyScanEncoder(): RocksDBDataEncoder = {
+    val (keySchema, _) = createTestSchemas()
+    createTestEncoder(PrefixKeyScanStateEncoderSpec(keySchema, numColsPrefixKey = 2))
+  }
+
+  private def createRangeKeyScanEncoder(): RocksDBDataEncoder = {
+    val (keySchema, _) = createTestSchemas()
+    createTestEncoder(RangeKeyScanStateEncoderSpec(keySchema, orderingOrdinals = Seq(0, 1)))
+  }
+
+  test("verify schema ID handling in prefix and range scan key encoding") {
+    val keySchema = StructType(Seq(
+      StructField("k1", IntegerType),
+      StructField("k2", LongType),
+      StructField("k3", DoubleType)
+    ))
+    val valueSchema = StructType(Seq(
+      StructField("v1", StringType)
+    ))
+
+    // Create test row with some data
+    val keyProj = UnsafeProjection.create(keySchema)
+    val fullKeyRow = keyProj.apply(InternalRow(42, 123L, 3.14))
+
+    // Test prefix scan encoding with schema evolution
+    {
```

Comment:
nit: do we need additional bracket while we also use withClue with code block? what's the usage? same for others.